### PR TITLE
Mark as received

### PR DESCRIPTION
**Pull request by cdolm92**
# What

Backers can now keep track of the rewards they've received in the backing page.

# See 👀

Snapshots below

# TODO ⏰

- [ ] Snapshot tests

### CODE REVIEW

File: Kickstarter-iOS/Views/Controllers/BackingViewController.swift [hunk 1]
```diff
@@ -187,9 +193,26 @@ internal final class BackingViewController: UIViewController {
     _ = self.loadingIndicatorView
       |> baseActivityIndicatorStyle
 
+    _ = [self.rewardDeliveredLabel, self.useThisToKeepTrackLabel]
+      ||> UILabel.lens.font .~ .ksr_body(size: 14)
+      ||> UILabel.lens.textColor .~ .ksr_dark_grey_900
+
+    _ = self.rewardDeliveredLabel
+      |> UILabel.lens.text %~ { _ in Strings.Reward_delivered() }
```
Comment (eoji) [inline Kickstarter-iOS/Views/Controllers/BackingViewController.swift]: This label should be set to bold.

File: KsApi/models/Backing.swift [hunk 2]
```diff
@@ -6,6 +6,8 @@ public struct Backing {
   public let amount: Int
   public let backer: User?
   public let backerId: Int
+  public let completed: Bool?
+  public let backerCompleted: Bool?
```
Comment (eoji) [inline KsApi/models/Backing.swift]: Let's change this to `backerCompletedAt` and make it a `Date` to match what we have on backend.

File: Library/ViewModels/BackingViewModel.swift [hunk 3]
```diff
@@ -16,6 +16,8 @@ public protocol BackingViewModelInputs {
 
   /// Call when the "View messages" button is pressed.
   func viewMessagesTapped()
+
```
Comment (eoji) [inline Library/ViewModels/BackingViewModel.swift]: Please add a comment here.

File: Library/ViewModels/BackingViewModel.swift [hunk 4]
```diff
@@ -109,9 +118,35 @@ public final class BackingViewModel: BackingViewModelType, BackingViewModelInput
...
+      .takePairWhen(self.rewardReceivedTappedProperty.signal)
+      .map(unpack)
+      .switchMap { project, backer, received in
+        AppEnvironment.current.apiService.backingUpdate(
+          forProject: project, forUser: backer, received: received)
+          .ksr_delay(AppEnvironment.current.apiDelayInterval, on: AppEnvironment.current.scheduler)
+          .materialize()
+    }
+
+    let markedReceived = rewardReceivedEvent.values().map { $0 }
```
Comment (eoji) [inline Library/ViewModels/BackingViewModel.swift]: Let's rename this to `markedReceivedBacking` for clarity.

File: Library/ViewModels/BackingViewModel.swift [hunk 5]
```diff
@@ -263,6 +305,19 @@ public final class BackingViewModel: BackingViewModelType, BackingViewModelInput
   public var outputs: BackingViewModelOutputs { return self }
 }
 
+private func shouldHideMarkReceived(backing: Backing, project: Project) -> Bool {
+
+  if !project.memberData.permissions.isEmpty {
```
Comment (eoji) [inline Library/ViewModels/BackingViewModel.swift]: We are wondering if you're a collaborator or creator and you backed your own project, we wouldn't necessarily want to hide this.

File: KsApi/models/Backing.swift [hunk 6]
```diff
@@ -25,6 +27,10 @@ public struct Backing {
     case pledged
     case preauth
   }
+
+  public var markedReceived: Bool {
```
Comment (eoji) [inline KsApi/models/Backing.swift]: Looks like this isn't used 🕵️‍♀️

File: Library/KSCache.swift [hunk 7]
```diff
@@ -8,6 +8,7 @@ public final class KSCache {
   public static let ksr_findFriendsFollowing = "find_friends_follow_view_model"
   public static let ksr_messageThreadHasUnreadMessages = "message_thread_has_unread_messages"
   public static let ksr_projectSaved = "project_saved"
+  public static let ksr_markedReceived = "marked_received"
```
Comment (eoji) [inline Library/KSCache.swift]: We didn't see this used anywhere. Do we need it?

**Comment by cdolm92** [comment]
g2g @eoji?

**Comment by eoji** [comment]
SHIP IT 🗣

**Pull request merged**